src/Polynomials.jl

```diff
@@ -348,14 +348,24 @@ end
 div(num::Poly, den::Poly) = divrem(num, den)[1]
 rem(num::Poly, den::Poly) = divrem(num, den)[2]
 
-function ==(p1::Poly, p2::Poly)
-    if p1.var != p2.var
-        return false
-    else
-        return p1.a == p2.a
-    end
+==(p1::Poly, p2::Poly) = (p1.var == p2.var && p1.a == p2.a)
+==(p1::Poly, n::Number) = (coeffs(p1) == [n])
+==(n::Number, p1::Poly) = (p1 == n)
+
+function isapprox{T,S}(p1::Poly{T}, p2::Poly{S};
+  rtol::Real = sqrt(eps(promote_type(T,S))), atol::Real = 0)
```

Comment:
... and should `isapprox` allowed to be throw as in `isapprox([1,2], [1,2,3])` (`DimensionMismatch`), or should it just return `false`? I am asking this, because the way I have implemented is allowed to throw. Then, you have problems in `pade`. But I can just return `false` by checking against `degree(p1) == 0` when it is with respect to `Number`s, or against `degree(p1) == degree(p2)` when it is between two `Poly`s. What do you think? Or, we can use `chop` and padd the vectors to have the same dimension, and check using `isapprox`.